sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousMessageSubscriber.java

```diff
@@ -307,6 +307,22 @@ private SynchronousReceiveWork getOrUpdateCurrentWork() {
                     .addKeyValue("numberOfEvents", work.getNumberOfEvents())
                     .log("Current work updated.");
 
+                //When work is completed, update next work as current work if work queue is not empty.
+                work.getCompleteEmitter().asMono()
+                    .subscribe(unused -> { },
+                        error -> {
+                            logger.atWarning()
+                                .addKeyValue(WORK_ID_KEY, work.getId())
+                                .log("Error occurred in current work. Schedule next work.", error);
+                            this.getOrUpdateCurrentWork();
+                        },
+                        () -> {
+                            logger.atVerbose()
+                                .addKeyValue(WORK_ID_KEY, work.getId())
+                                .log("Current work completed. Schedule next work.");
+                            this.getOrUpdateCurrentWork();
```

Comment:
Curious on why previous code works. If no one call `getOrUpdateCurrentWork()`, the 2nd work in queue won't get in to current work?
The other 2 places it get called is `hookOnSubscribe` and `queueWork`.

Reply:
> If no one call getOrUpdateCurrentWork(), the 2nd work in queue won't get in to current work
It seems right. I try to start 10 threads to receive messages at same time, and only the 1st work got finished - using v.7.6.0. 

However, in this scenario, current fix also met problem when starting 2nd work, the issue is related to the credit calculation. I will take a further look at this.  Now the fix only works for one thread start receiving message at one time.